### PR TITLE
Migrate sig-apimachinery and sig-servicecatalog e2e tests

### PR DESCRIPTION
**Pull request by guangxuli**
**What this PR does / why we need it**:
Migrate sig-apimachinery and sig-servicecatalog e2e tests

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Ref Umbrella issue #49161
1. Move generated_clientset.go to sig-apimachinary
2. Move podpreset.go to sig-servicecatalog by creating new directory.

**Special notes for your reviewer**:

**Release note**:

none

/cc @liggitt

### CODE REVIEW

File: test/e2e/apimachinery/generated_clientset.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package e2e
+package apimachinary
```
Comment (liggitt) [inline test/e2e/apimachinery/generated_clientset.go]: apimachinery

File: test/e2e/servicecatalog/podpreset.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package e2e
+package servicecatalog
```
Comment (liggitt) [inline test/e2e/servicecatalog/podpreset.go]: you have to add `_ "k8s.io/kubernetes/test/e2e/servicecatalog"` to test/e2e/e2e_test.go or this test won't be included/run in the test suite

File: test/e2e/apimachinery/generated_clientset.go (hunk 1 above)
Comment (guangxuli) [inline test/e2e/apimachinery/generated_clientset.go]: oh, yes, addressed.

File: test/e2e/servicecatalog/podpreset.go (hunk 2 above)
Comment (guangxuli) [inline test/e2e/servicecatalog/podpreset.go]: Done, thanks @liggitt

File: test/e2e/servicecatalog/podpreset.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-var _ = framework.KubeDescribe("PodPreset", func() {
+var _ = SIGDescribe("PodPreset", func() {
```
Comment (liggitt) [inline test/e2e/servicecatalog/podpreset.go]: Does SIGDescribe need to be implemented for this package?

Comment (guangxuli) [inline test/e2e/servicecatalog/podpreset.go]: yes, indeed, I just push a framework file that implement the SIGDescribe.

**Comment by guangxuli** [comment]
@liggitt any comments?

**Comment by guangxuli** [comment]
thanks @liggitt

**Comment by guangxuli** [comment]
ping @fabianofranz @deads2k PTAL, thanks.

**Pull request merged**